source/ShipJumpNavigation.h

```diff
@@ -0,0 +1,97 @@
+/* ShipJumpNavigation.h
+Copyright (c) 2022 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef SHIP_JUMP_NAVIGATION_H_
+#define SHIP_JUMP_NAVIGATION_H_
+
+#include "JumpTypes.h"
+
+#include <map>
+
+class Outfit;
+class Ship;
+class System;
+
+
+
+// A class representing the jump capabilities of a ship. Calculates and caches a ship's
+// jump methods, costs, and distances.
+class ShipJumpNavigation {
+public:
+	static const double DEFAULT_HYPERDRIVE_COST;
+	static const double DEFAULT_SCRAM_DRIVE_COST;
+	static const double DEFAULT_JUMP_DRIVE_COST;
+
```

Comment:
I think we still do two empty lines even though the access is the same between the above and below.
```suggestion


```